### ISSUE TITLE
Generate DockerFile

### ISSUE THREAD

**Issue by seasuresh** [GoogleContainerTools/jib#2394]
I'm unable to view DockerFile generated locally. Is it possible to generate DockerFile so we can push the image manually to the repo or issuing one more command?

**Comment by seasuresh** [comment]
Is there any alternative to user ``jibExportDockerContext`` on the latest version?

**Comment by loosebazooka** [comment]
That feature was deprecated a while ago. It was not possible to accurately recreate what jib was doing by using a docker file, however see FAQ: https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#what-would-a-dockerfile-for-a-jib-built-image-look-like

**Comment by loosebazooka** [comment]
What are you trying to achieve? We have tasks/goals like `jibDockerBuild` that can build to a docker daemon. Or jibBuildTar that builds a file you can `docker load` into a docker daemon.

**Comment by seasuresh** [comment]
I'm integrating the spring boot to deploy via Concourse. As part of this deployment, we use the below Concourse resource to push the DockerFile to AWS ECR. We build and push to lower the environment and use the same file to push it to the remaining environments. If i use jib to push to one environment the below resource doesn't recognize the version. The possible options are 

1) Build and deploy every environment using jib command
2) Downgrade jib to the older version to generate DockerFile and let the below Concourse resource to push the image.

It would be great if the newer versions of the jib to generate DockerFile and context 

https://github.com/concourse/docker-image-resource

**Comment by loosebazooka** [comment]
So I'm not familiar with concourse but it really seems strange to send a dockerfile around - implying that you are doing multiple builds. Can't you do a build once using jib and send an image reference around?

There are some statements you've included in your issue that I do not understand
> We build and push to lower the environment...

> ...the below resource doesn't recognize the version

And so it's hard to follow what your exact issue is. From the documentation it seems like `docker-image-resource` maybe is designed specifically for docker based builds (which jib is not) -- jib is a container builder that specifically _does not_ use docker. is specifically Can you elaborate more and include every detail?

Is there not a way to specify custom jobs for jib within concourse?

Are you specifically tied to concourse? We have pretty deep integration for jib in [skaffold](https://github.com/GoogleContainerTools/skaffold).

**Comment by loosebazooka** [comment]
> It would be great if the newer versions of the jib to generate DockerFile and context

Also regarding this. We have no plans for this, there was too much deviation between what jib was generating and what the dockerfile was generating.  And beyond that, the tool in the FAQ can do what you're asking: https://github.com/gclayburg/dockerPreparePlugin
